ruby.c

```diff
@@ -1407,6 +1408,18 @@ proc_long_options(ruby_cmdline_options_t *opt, const char *s, long argc, char **
     else if (is_option_with_arg("external-encoding", Qfalse, Qtrue)) {
         set_external_encoding_once(opt, s, 0);
     }
+    else if (is_option_with_arg("parser", Qfalse, Qtrue)) {
+        if (strcmp("prism", s) == 0) {
+            (*rb_ruby_prism_ptr()) = true;
+            rb_warn("The Prism compiler is currently experimental and compatibility with parse.y is not yet complete. Please report an issues you find on the prism issue tracker.");
```

Comment:
Can we not compare the "Prism compiler" to "parse.y" please? It is the "Prism parser" that is comparable to "parse.y", not the compiler based on that parse tree.

How does the following read?
```suggestion
            rb_warn("The compiler based on the Prism parser is currently experimental and compatibility with the current Ruby compiler is not yet complete. Please report any issues you find on the `ruby/prism` issue tracker.");
```

Reply:
I feel like this message is a bit more complicated this way, but I agree it was inaccurate about which bits compare with each other.